test/kubemark/common.sh

```diff
@@ -39,31 +39,33 @@ MASTER_NAME="${INSTANCE_PREFIX}-kubemark-master"
 MASTER_TAG="kubemark-master"
 EVENT_STORE_NAME="${INSTANCE_PREFIX}-event-store"
 
-RETRIES=3
-
 export KUBECTL="${KUBE_ROOT}/cluster/kubectl.sh"
 export KUBEMARK_DIRECTORY="${KUBE_ROOT}/test/kubemark"
 export RESOURCE_DIRECTORY="${KUBEMARK_DIRECTORY}/resources"
 
 # Runs gcloud compute command with the given parameters. Up to $RETRIES will be made
 # to execute the command.
 # arguments:
-# $@: all stuff that goes after 'gcloud compute '
+# $@: all stuff that goes after 'gcloud compute'
 function run-gcloud-compute-with-retries {
-  echo "" > /tmp/gcloud_retries
+  RETRIES="${RETRIES:-3}"
   for attempt in $(seq 1 ${RETRIES}); do
-    if ! gcloud compute $@ &> /tmp/gcloud_retries; then
-      if [[ $(grep -c "already exists" /tmp/gcloud_retries) -gt 0 ]]; then
+    local -r gcloud_cmd_hash=`echo "gcloud compute $@" | md5sum | cut -f1 -d" "`
+    local -r gcloud_logfile="/tmp/gcloud_${gcloud_cmd_hash}.log"
+    echo "" > ${gcloud_logfile}
+    if ! gcloud compute "$@" |& tee ${gcloud_logfile}; then
+      if [[ $(grep -c "already exists" ${gcloud_logfile}) -gt 0 ]]; then
```

Comment:
I think this is overcomplicated. Why we just can't assign the result of gcloud command to the local env var?